colexec: remove batch width check in invariant checker

Azhng opened the pull request:
The batch width check in invariant checker is removed since
this assumption is no longer true. This fixes the issue that
ordinality operator fails the batch width check in logictest
when batch size is 1.

Release justification: bug fix
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/46381)
<!-- Reviewable:end -->

Azhng commented:
Benchmark result 

```
name          old time/op    new time/op    delta
Ordinality-8    2.74µs ± 4%    2.86µs ± 2%  +4.59%  (p=0.000 n=10+8)

name          old speed      new speed      delta
Ordinality-8  2.99GB/s ± 4%  2.86GB/s ± 2%  -4.41%  (p=0.000 n=10+8)
```

asubiotto commented:
Could you share how setting the batch size to 1 resulted in triggering this bug?

asubiotto commented:
I like the solution of the ordinality operator owning the returned batch and appending a column to it rather than the input batch, but I'd like @yuzefovich to take a look since he's done a lot of thinking about batch widths

yuzefovich reviewed (commented):
I don't like this solution because it makes ordinality operator special, but we have other operators that append columns and they do not do this. Is the bug fix for the failure of "invariants checker"? I think the correct approach is to fix the checker.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/46381#-:-M36r0mvCx4ib9Wi5Xnq:b-qmxr06)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @yuzefovich)



<!-- Sent from Reviewable.io -->

Azhng committed:
colexec: remove batch width check in invariant checker

The batch width check in invariant checker is removed since
this assumption is no longer true. This fixes the issue that
ordinality operator fails the batch width check in logictest
when batch size is 1.

Release justification: bug fix
Release note: None

Azhng reviewed (commented):
Updated to remove batch width check in the invariants checker 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/46381#-:-M37RtxX9ZHRqRlQOkX5:b3ajbvr)* status: :shipit: complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

yuzefovich reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

nit: PR title and description should be updated.

> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/46381#-:-M37mH5W27PawvoBujbM:b-jfsuuw)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

Azhng commented:
TFTR!

bors r+

Azhng commented:
bors r+

Azhng commented:
bors r+
